src/error-code.ts

```diff
@@ -686,6 +686,76 @@ export namespace ErrorCodes {
       type: ErrorType.ERROR,
       title: "Unable to find reference out of application function",
     };
+
+  /**
+   * Appsync Integration invocations must be deterministic.
+   *
+   * As stated in the [AppSync Pipeline Resolvers Documents](https://docs.aws.amazon.com/appsync/latest/devguide/pipeline-resolvers.html):
+   *
+   * > Pipeline resolver execution flow is unidirectional and defined statically on the resolver.
+   *
+   * ```ts
+   * const func = new Function(stack, 'id', async () => {});
+   *
+   * new AppsyncResolver(..., async ($context) => {
+   *    // valid
+   *    const f = await func();
+   *    // valid
+   *    await func();
+   *    if($context.arguments.value) {
+   *       // invalid - non-deterministic
+   *       await func();
+   *    }
+   *    while($context.arguments.value) {
+   *       // invalid
+   *       await func();
+   *    }
+   *    // valid
+   *    return func();
+   * });
+   * ```
+   *
+   * Workaround:
+   *
+   * One workaround would be to invoke a lambda function which handles the non-deterministic parts of the workflow.
+   *
+   * The result of this example would be to call the `conditionalFunc` deterministically and call `func` conditionally.
+   *
+   * ```ts
+   * const func = new Function(stack, 'id', async () => {});
+   *
+   * const conditionalFunc = new Function(stack, 'id', async (value) => {
+   *    if(value) {
+   *       return func();
+   *    }
+   *    return null;
+   * })
+   *
+   * new AppsyncResolver(..., async ($context) => {
+   *    const conditionalResult = await conditionalFunc();
+   *    return conditionalResult;
+   * });
+   * ```
+   */
+  export const Appsync_Integration_invocations_must_be_deterministic: ErrorCode =
+    {
+      code: 10020,
+      type: ErrorType.ERROR,
+      title: "Appsync Integration invocations must be deterministic",
+    };
+
+  /**
+   * Unsupported Feature
+   *
+   * Generic error for unsupported features.
+   *
+   * See error message provided for more details.
+   */
+  export const Unsupported_Feature: ErrorCode = {
+    code: 10021,
+    type: ErrorType.ERROR,
+    title: "Unsupported feature",
+  };
```

Comment:
Let's remove this error and instead keep the discipline or defining an error code for each case. The message is not helpful, which will bloat documentation of error codes, and the code that throws and error will still have to provide a description. Let's keep it consistent.

Reply:
I didn't like creating a detailed error codes for features we need to obviously support. Adding a message with an issue link isn't hard to do inline when throwing. 

Having an error for each missing feature will bloat the error codes more than having a single error code with a replacement error message. 

```ts
ErrorCodes.For_Loops_Are_Unsupported
ErrorCodes.Switch_Statement_Are_Unsupported
ErrorCodes.For_Loop_Expressions_Initialization_Is_Unsupported
```

If we don't have a generic error message, we'll more likely just throw useless errors with no description. 

I'd rather spend the time cutting the ticket and implementing the feature than pruning a bunch of transient error codes. (I'd also like to not throw cryptic, unactionable errors). 

Proposal: 
* If it is a system limitation, a large feature, or a edge case we will likely not support for a while - Add an error code
* If it is a common use case that we have not gotten to implementing - add unsupported feature with a link to a ticket to resolve 
* If it is a case we don't expect to happen, use Unexpected Error with a description of why it shouldn't happen.